tests/JournalIntegrationTests.js

```diff
@@ -0,0 +1,299 @@
+            args: ["{kettle.test.configuration}", "{that}", "{arguments}.0"]
+        }
+    }
+});
+
+// Reach upwards into the global configuration's server and destroy it
+gpii.tests.journal.settingsHandlerExplode = function (configuration, that, payload) {
+    fluid.log("EXPLODING SETTINGS HANDLER EXECUTING with id " + that.id + " and payload ", payload);
+    // null out our payload, compensate for GPII-1223/GPII-1891
+    payload["com.microsoft.windows.cursors"][0].settings = {};
+    // Beat jqUnit's failure handler to ignore the various errors falling out from this process
+    // kettle.test.pushInstrumentedErrors(function () {
+    //    fluid.log("Received expected failure from destroying active server: ", arguments);
+    //});
+    fluid.invokeLater(function () { // invoke later so that we do not race with construction - TODO: framework bug
+        configuration.server.destroy();
+        fluid.log("Server destroyed");
+    });
+    // TODO: failure of Ungarism here - invokers need to be eliminated from the framework
+    return gpii.settingsHandlers.invokeSettingsHandler(that.setImpl, payload);
+};
+
+gpii.tests.journal.populateExplodingSettingsHandler = function (that) {
+    gpii.test.integration.mockSettingsHandlerRegistry.populateOne(that, that.explodingSettingsHandler, "gpii.tests.journal.explodingSettingsHandler");
+};
+
+gpii.tests.journal.logDestroyed = function (loginRequest, testCaseHolder) {
+    fluid.log("Received expected server destroyed event");
+    // Remove KettleTestUtils.http native handler to this which will trigger a test failure
+    loginRequest.nativeRequest.removeAllListeners("error");
+    loginRequest.nativeRequest.on("error", function (err) {
+        fluid.log("Received expected hangup error from login request", err);
+    });
+    jqUnit.assert("Reached checkpoint for destruction of server");
+    var mocks = fluid.queryIoCSelector(fluid.rootComponent, "gpii.test.integration.mockSettingsHandler", true);
+    fluid.each(mocks, function (mock) {
+        mock.settingsStore = {};
+    });
+    fluid.log("CLEARED " + mocks.length + " mock settingsHandlers");
+    // Arbitrarily wait as a result of FLUID5931 bug preventing immediate reconstruction of the server
+    fluid.invokeLater(testCaseHolder.events.FLUID5931wait.fire);
+};
+
+gpii.tests.journal.stashJournalId = function (component) {
+    component.stashedStartTime = Date.now();
+    component.stashedJournalId = ">" + new Date(component.stashedStartTime).toISOString();
+};
+
+gpii.tests.journal.checkJournalsList = function (markup, component) {
+    var match = /a href=".*\/%3E(.*)"/.exec(markup);
+    var firstDate = decodeURIComponent(match[1]);
+    var firstTime = Date.parse(firstDate);
+    fluid.log("Parsed link date " + firstDate + " to time" + firstTime);
+    jqUnit.assertTrue("Received correct journal time in journal list markup", firstTime > component.stashedStartTime && firstTime < (component.stashedStartTime + 2000));
+};
```

Comment:
I'm thinking that we could expand the JournalIntegrationTests.js a little to include both the crashed and regular log out cases. We could then verify both the "closed" value recorded in the journal file and the labeling on the HTML journal list.